raft-tech/TANF-app
# Issue 767: Logging Client Alerts and Errors

### Summary of Changes

This PR adds a generic logging endpoint to the backend to which the client sends errors and alerts.

![Screen Shot 2021-04-27 at 4 30 33 PM](https://user-images.githubusercontent.com/12901172/116308843-02ce8300-a776-11eb-9302-cf615b6f3266.png)

_Addresses issue #767
_Acceptance criteria as stated in the issue_

### How to Test
_List the steps to test the PR_

- Run the frontend and backend apps locally
- Click the Data Files tab
- Click Search
- Add a valid file to any file input
- Click Submit Data Files
- A request should be sent to the backend `/logs` endpoint, and this can be confirmed in the browser's network tab
- In the terminal where the backend is running, the log sent from the frontend should be visible

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

	- [x] Endpoint for logging messages from the client exists
    - [x] Client is updated to send messages to the endpoint on errors and alerts 
    - [x] Screenshot of logged alerts/errors is posted to the pull request

As facilitator/product manager, @kniz-raft will decide if ACs are met from Raft's perspective.

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*


### Deliverable 4: Accessible



Does not apply.

### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful build with a single command*

> NOTE: until we have a proper staging environment this may not be satisfiable prior to merging


### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*


### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*

  - None detected
